### PR TITLE
kde-neon-6: Fix long standing theming issues with Qt snaps.

### PR DESCRIPTION
**Pull request by ScarlettGatelyMoore**
- [x ] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x ] Have you successfully run `tox run -m lint`?
- [ x] Have you successfully run `tox run -e test-py310`? (supported versions: `py39`, `py310`, `py311`, `py312`)

-----
Tests pass, there will also come with this a qt-common-themes snap that will hold qt themes and our breeze gtk themes which the consumers will have to connect until I get an auto connect. I will also have to do a MR to remove the non-functional outdated breeze from gtk-common-themes.

### CODE REVIEW

**Review by lengau — APPROVED** [review]

File: tests/unit/extensions/test_kde_neon_6.py [hunk 1]
```diff
@@ -221,12 +291,8 @@ def assert_get_part_snippet(kde_neon_6_instance):
                 {
                     "LD_LIBRARY_PATH": (
                         "/snap/kde-qt6-core22-sdk/current/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}:"
-                        "/snap/kde-qt6-core22-sdk/current/usr/lib:"
                         "/snap/kf6-core22-sdk/current/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}:"
-                        "/snap/kf6-core22-sdk/current/usr/lib/${CRAFT_ARCH_TRIPLET_BUILD_FOR}"
```
Comment (sergiusens) [inline tests/unit/extensions/test_kde_neon_6.py:226]: Aside from this being a very very welcomed PR :tada:, is there a reason to remove these?

Comment (ScarlettGatelyMoore) [reply tests/unit/extensions/test_kde_neon_6.py:226]: Well gosh, something went terribly wrong. I don't have these changes in my local branch. Did I somehow miss a merge? How would it let me.

**Review by mr-cal — APPROVED** [review]
Thanks!

Do you have a link to any issue or bug reports associated with this?  I'd like to include some sort of reference in the changelog.

**Comment by ScarlettGatelyMoore** [comment]
> Thanks!
> 
> Do you have a link to any issue or bug reports associated with this? I'd like to include some sort of reference in the changelog.

https://forum.snapcraft.io/t/icons-theming-and-font-issues-in-qt-snaps-lubuntu/33853
https://forum.snapcraft.io/t/qt-app-and-qt-and-kvantum-themes/15234
https://forum.snapcraft.io/t/how-to-fix-weird-cursor-theme-and-fonts-in-qt-apps-on-elementary-or-other-distribution-solution/10248
https://forum.snapcraft.io/t/theme-issue-with-content-snap/13005
https://bugs.launchpad.net/snappy/+bug/1836054
https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1585332

Just to name a few...

**Comment by mr-cal** [comment]
This needs more auto-connect permissions, see discussion in the forum: https://forum.snapcraft.io/t/kde-global-auto-connect-qt-common-themes/40878

**Comment by ScarlettGatelyMoore** [comment]
> This needs more auto-connect permissions, see discussion in the forum: https://forum.snapcraft.io/t/kde-global-auto-connect-qt-common-themes/40878

I don't follow. They rejected my requests and told me to make snaps for our themes. I do need to update the extension, but I won't be able to get to it for another week or so.

**Comment by mr-cal** [comment]
Sorry, that was a general note to link to that forum request, not a request for you.

I'm trying to decide if this should be in Snapcraft 8.4 or delayed until 8.5.

**Comment by ScarlettGatelyMoore** [comment]
> Sorry, that was a general note to link to that forum post, not a request for you.
> 
> I'm trying to decide if this should be in Snapcraft 8.4 or delayed until 8.5.

I would delay it as I don't have qt themes working. The gtk themes mostly work as snaps.

**Comment by mr-cal** [comment]
Sounds good, thanks for your opinion.

I'll plan to revert this in a `hotfix/8.4` branch and leave it in `main`.

**Pull request merged**